website/static/js/koHelpers.js

```diff
@@ -459,6 +459,50 @@ ko.bindingHandlers.datePicker = {
 };
 
  /**
+ * Bind content of contenteditable to observable. Looks for maxlength attr
+ * and underMaxLength binding to limit input.
+ * Example:
+ * <div contenteditable="true" data-bind="editableHTML: <observable_name>" maxlength="500"></div>
+ */
+ko.bindingHandlers.editableHTML = {
+    init: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        $element.html(initialValue);
+        $element.on('change input paste keyup blur', function() {
+            var observable = valueAccessor();
+            observable($element.html());
+        });
+    },
+    update: function(element, valueAccessor, allBindings, bindingContext) {
+        var $element = $(element);
+        var initialValue = ko.utils.unwrapObservable(valueAccessor());
+        var charLimit = $element.attr('maxlength');
+        if (charLimit && bindingContext.underMaxLength) {
+            var inputTextLength = $element[0].innerText.length;
+            var errorExceedsCharacterLimit = 'Exceeds character limit. Please reduce to ' +
+                charLimit + ' characters or less.';
+            // + 1 to account for the <br> that is added to the end of the contenteditable content
+            // <br> is necessary for the return key to function properly
+            if (inputTextLength > parseInt(charLimit) + 1) {
+                bindingContext.underMaxLength(false);
```

Comment:
This makes this binding handler difficult to reuse because it assumes a lot about the ViewModel that controls it (it requires `underMaxLength` and `errorMessage` to be observables on the ViewModel).

How about passing in an error handler function?

```html
<div contenteditable="true" 
    data-bind="editableHTML: {observable: comment, onError: handleError" maxlength="500">
</div>
```

```js
var ViewModel = function(...) {
    var self = this;
    self.content = ko.observable();
    self.underMaxLength(true);
    self.errorMessage = ko.observable();
    self.handleError = function(message) {
          self.errorMessage(message);
          self.underMaxLength(false);
     }
}
```

Not a blocker, but would be a nice refactor.